locksmith/__tests__/operations/userOperations.test.ts

```diff
@@ -0,0 +1,86 @@
+import UserOperations from '../../src/operations/userOperations'
+
+const models = require('../../src/models')
+let User: any = models.User
+let UserReference: any = models.UserReference
+
+describe('User creation', () => {
+  let userCreationDetails = {
+    emailAddress: 'USER@EXAMPLE.COM',
+    publicKey: '0x21cc9c438d9751a3225496f6fd1f1215c7bd5d83',
+    passwordEncryptedPrivateKey: '{"data" : "encryptedPassword"}',
+    recoveryPhrase: 'recoveryPhrase',
+  }
+
+  describe("when a user for the public key doesn't exist", () => {
+    it('should create a user and associated data', async () => {
+      UserReference.create = jest.fn(() => {})
+
+      expect.assertions(1)
+      await UserOperations.createUser(userCreationDetails)
+      expect(UserReference.create).toHaveBeenCalledWith(
+        {
+          User: {
+            passwordEncryptedPrivateKey: '{"data" : "encryptedPassword"}',
+            publicKey: '0x21cC9C438D9751A3225496F6FD1F1215C7bd5D83',
+            recoveryPhrase: 'recoveryPhrase',
+          },
+          emailAddress: 'user@example.com',
+        },
+        { include: User }
+      )
+    })
+
+    it('should normalize the public key/address & email address', async () => {
```

Comment:
Cpuld you add a test where it returns `false` if the we try to create the same user twice?

Reply:
A similar sentiment to the other test above

Reply:
This logic is tested in the controller's tests